src/stcal/jump/jump.py

```diff
@@ -459,15 +507,110 @@ def find_ellipses(dqplane, bitmask, min_area):
-                if jump not in snowballs:
-                    snowballs.append(jump)
-    return snowballs
+    for jump in jump_ellipses:
+        sat_found = False
+        if near_edge(jump, low_threshold, high_threshold):
+            snowballs.append(jump)
+        else:
+            for sat in sat_ellipses:
+                # center of saturation is within the enclosing jump rectangle
+                if point_inside_ellipse(sat[0], jump):
+                    if jump not in snowballs:
+                        snowballs.append(jump)
+                        sat_found = True
+                        gdq[integration, :, :, :] = extend_saturation(gdq[integration, :, :, :],
+                                                                      group, [sat], sat_flag, jump_flag,
+                                                                      min_sat_radius, expansion)
+    return gdq, snowballs
+
+
+def point_inside_ellipse(point, ellipse):
+    delta_center = np.sqrt((point[0]-ellipse[0][0])**2 + (point[1]-ellipse[0][1])**2)
+    minor_axis = min(ellipse[1][0], ellipse[1][1])
+    if delta_center < minor_axis:
+        return True
+    else:
+        return False
+
+#  This routing tests whether the center of a jump is close to the edge of the detector.
+#  Jumps that are within the threshold will not requre a saturated core since this may be off the
+#  detector
+def near_edge(jump, low_threshold, high_threshold):
+    if jump[0][0] < low_threshold or jump[0][1] < low_threshold\
+        or jump[0][0] > high_threshold or jump[0][1] > high_threshold:
+        return True
+    else:
+        return False
+
+def find_faint_extended(data, gdq, read_noise_2d, nframes, snr_threshold=1.3, min_shower_area=40, inner=1,
+                            outer=2, sat_flag=2, jump_flag=4, ellipse_expand = 1.1, num_grps_masked=25):
+    """
+    :param data: The input data cube
+    :param gdq: The group dq cube after regular jump detection has been run
+    :param read_noise_2d: The read noise for each pixel
+    :param nframes: The number frames that are averaged in the group
+    :param snr_threshold: The signal-to-noise ratio threshold for detection of extended emission
+    :param min_shower_area: The minimum area for a group of pixels to be flagged as a shower
+    :param inner: The inner radius of the ring_2D_kernal used for the convolution
+    :param outer: The outer radius of the ring_2D_kernal used for the convolution
+    :param sat_flag: The integer value of the saturation flag
+    :param jump_flag: The integer value of the jump flag
+    :param ellipse_expand: The relative increase in the size of the fitted ellipse to be applied to the shower
+    :param num_grps_masked: The number of groups after the detected shower to be flagged as jump
+    :return: gdq (updated gdq cube)
+             number_ellipse (total number of showers detected)
+    """
+    read_noise_2 = read_noise_2d**2
+    data[gdq == sat_flag] = np.nan
```

Comment:
nans are being set here in the data, should make sure this is operating on a copy or use a mask throughout

Reply:
I agree this is a valid concern. I don't think it would break anything but the uncal file would have a lot of NaNs after pixels saturated. I made a copy of the incoming data.